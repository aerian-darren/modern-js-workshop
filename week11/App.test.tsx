// import * as React from "react";
// import * as ReactDOM from "react-dom";
// import * as renderer from "react-test-renderer";
// import { App } from "./App";

// it("renders without crashing", () => {
//     const run = () => {
//         const div = document.createElement("div");
//         ReactDOM.render(<App />, div);
//         ReactDOM.unmountComponentAtNode(div);
//     };
//     expect(run).not.toThrow();
// });

// it("renders all the components we need", () => {
//     const tree = renderer.create(<App />).toJSON();
//     let treeChildrenLength = 0;

//     if (tree && tree.children) {
//         treeChildrenLength = tree.children.length;
//     }

//     expect(treeChildrenLength).toBeGreaterThanOrEqual(2);
//     expect(tree).toMatchSnapshot();
// });

// it("removes an emoji from the end of the output when delete is pressed", () => {
//     const state: State = {
//         output: ["Default output"],
//         emojilist: [],
//         categories: [],
//         filteredEmojis: [],
//         selectedCategory: "",
//     };

//     const props: Props = {};
//     const app = new App(state, props);

//     app.setState = jest.fn();
//     app.addEmoji("ðŸ˜»");
//     app.removeEndEmoji();

//     expect(app.setState).toHaveBeenCalledWith({
//         output: ["Default output"],
//     });
// });
