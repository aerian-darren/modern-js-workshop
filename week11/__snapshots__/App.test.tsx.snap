// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all the components we need 1`] = `
<div
  className="typewriter"
>
  <h1>
    Type your emoji
  </h1>
  <select
    onChange={[Function]}
  >
    <option
      value="objects"
    >
      objects
    </option>
    <option
      value="faces"
    >
      faces
    </option>
    <option
      value="gestures"
    >
      gestures
    </option>
    <option
      value="transportation"
    >
      transportation
    </option>
    <option
      value="people"
    >
      people
    </option>
  </select>
  <button
    onClick={[Function]}
  >
    Clear
  </button>
  <div
    id="keyboard"
  >
    <button
      onClick={[Function]}
    >
      🎿
    </button>
    <button
      onClick={[Function]}
    >
      🏷
    </button>
    <button
      onClick={[Function]}
    >
      👑
    </button>
    <button
      onClick={[Function]}
    >
      👕
    </button>
    <button
      onClick={[Function]}
    >
      👖
    </button>
    <button
      onClick={[Function]}
    >
      👗
    </button>
    <button
      onClick={[Function]}
    >
      👛
    </button>
    <button
      onClick={[Function]}
    >
      👝
    </button>
    <button
      onClick={[Function]}
    >
      👢
    </button>
    <button
      onClick={[Function]}
    >
      💊
    </button>
    <button
      onClick={[Function]}
    >
      💍
    </button>
    <button
      onClick={[Function]}
    >
      💎
    </button>
    <button
      onClick={[Function]}
    >
      💴
    </button>
    <button
      onClick={[Function]}
    >
      💶
    </button>
    <button
      onClick={[Function]}
    >
      💷
    </button>
    <button
      onClick={[Function]}
    >
      💺
    </button>
    <button
      onClick={[Function]}
    >
      📀
    </button>
    <button
      onClick={[Function]}
    >
      📅
    </button>
    <button
      onClick={[Function]}
    >
      📔
    </button>
    <button
      onClick={[Function]}
    >
      📖
    </button>
    <button
      onClick={[Function]}
    >
      📚
    </button>
    <button
      onClick={[Function]}
    >
      📟
    </button>
    <button
      onClick={[Function]}
    >
      📠
    </button>
    <button
      onClick={[Function]}
    >
      📣
    </button>
    <button
      onClick={[Function]}
    >
      📻
    </button>
    <button
      onClick={[Function]}
    >
      📼
    </button>
    <button
      onClick={[Function]}
    >
      🔑
    </button>
    <button
      onClick={[Function]}
    >
      🔒
    </button>
    <button
      onClick={[Function]}
    >
      🔔
    </button>
    <button
      onClick={[Function]}
    >
      🕳
    </button>
    <button
      onClick={[Function]}
    >
      🖪
    </button>
    <button
      onClick={[Function]}
    >
      🖫
    </button>
    <button
      onClick={[Function]}
    >
      🖺
    </button>
    <button
      onClick={[Function]}
    >
      🗈
    </button>
    <button
      onClick={[Function]}
    >
      🗏
    </button>
    <button
      onClick={[Function]}
    >
      🗐
    </button>
    <button
      onClick={[Function]}
    >
      🗿
    </button>
    <button
      onClick={[Function]}
    >
      🚪
    </button>
    <button
      onClick={[Function]}
    >
      🛀
    </button>
    <button
      onClick={[Function]}
    >
      ⌚️
    </button>
  </div>
  <h1>
    Type your emoji
  </h1>
</div>
`;
